pkg/decoders/base64.go

```diff
@@ -129,12 +146,38 @@ func getSubstringsOfCharacterSet(data []byte, threshold int, charsetMapping [128
 	return substrings
 }
 
+// appendB64Substring processes and appends a base64 substring to the result slice.
+// It handles both standard and URL-safe base64 encodings, preserving padding characters
+// and special characters like underscores.
 func appendB64Substring(data []byte, start, count int, substrings []string, endChars string) []string {
-	substring := bytes.TrimLeft(data[start:start+count], endChars)
-	if idx := bytes.IndexByte(bytes.TrimRight(substring, endChars), '='); idx != -1 {
-		substrings = append(substrings, string(substring[idx+1:]))
-	} else {
-		substrings = append(substrings, string(substring))
+	left := start
+	right := start + count - 1
+
+	// Trim left end chars except underscore.
+	for left < right && bytes.IndexByte([]byte(endChars), data[left]) != -1 && data[left] != '_' {
+		left++
+	}
+
+	// Trim right end chars except underscore and equals.
+	for right > left && bytes.IndexByte([]byte(endChars), data[right]) != -1 {
+		if data[right] == '=' || data[right] == '_' {
+			break
+		}
+		right--
+	}
+
+	// Find equals sign in the middle if present.
+	for i := left; i < right; i++ {
+		if data[i] == '=' {
+			left = i + 1
+			break
+		}
+	}
```

Comment:
This finds the first `=` between `left` and `right` and sets `left` to be one past it. This would be equivalent, correct?

```go
if equalPos := bytes.IndexByte(data[left:right], '='); equalPos != -1 {
    left = equalPos + 1
}
```